### PR TITLE
Log stuff to upper staff when set or removed

### PR DESCRIPTION
**Pull request by ghost**
Should Prevent the double setting of custom colors and/or double removing

Needs testing!

### CODE REVIEW

**Comment by ghost** [comment]
Also, please tell me if there are any more commands I need to log to prevent the double setting

**Comment by HoeenCoder** [comment]
Custom avatar, icon (might alredy be done)

**Comment by ghost** [comment]
I spoke with Kraken Mare, basically everything that's in the shop :v

File: spacialgaze-plugins/colors.js [hunk 1]
```diff
@@ -63,9 +63,9 @@ exports.commands = {
 			for (let u = 0; u < target.length; u++) target[u] = target[u].trim();
 			if (!target[1]) return this.parse('/help customcolor');
 			if (toId(target[0]).length > 19) return this.errorReply("Usernames are not this long...");
-
 			this.sendReply("|raw|You have given <b><font color=" + target[1] + ">" + Chat.escapeHTML(target[0]) + "</font></b> a custom color.");
 			this.privateModCommand("(" + target[0] + " has recieved custom color: '" + target[1] + "' from " + user.name + ".)");
+			Monitor.adminLog(target[0] + " has recieved custom color: '" + target[1] + "' from " + user.name + ".");
```
Comment (Mystifi) [inline spacialgaze-plugins/colors.js]: Typo: it should be spelt as "received"

**Comment by Mystifi** [comment]
This looks fine, and should work.

File: spacialgaze-plugins/colors.js (hunk 1 above)
Comment (ghost) [reply spacialgaze-plugins/colors.js]: I just use copy/paste ~~blame Hoeen~~

**Comment by DeathlyPlays** [comment]
Sounds like a good idea to me

**Comment by HoeenCoder** [comment]
Log who made the changes too.

**Comment by ghost** [comment]
Ok Will do... Eventually (hopefully)

**Comment by HoeenCoder** [comment]
Missing custom titles and icons

**Comment by ghost** [comment]
Sorry, darth proacrastination holds me captive

**Comment by ghost** [comment]
This should be everything!

File: spacialgaze-plugins/icons.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ exports.commands = {
 			if (toId(target[0]).length > 19) return this.errorReply("Usernames are not this long...");
 			if (icons[toId(target[0])]) return this.errorReply("This user already has a custom userlist icon.  Do /icon delete [user] and then set their new icon.");
 			this.sendReply("|raw|You have given " + SG.nameColor(target[0], true) + " an icon.");
-			Rooms('staff').add('|raw|' + SG.nameColor((target[0])) + ' has received an icon from ' + SG.nameColor(user.name), true).update();
+			Rooms('upperstaff').add('|raw|' + SG.nameColor((target[0])) + ' has received an icon from ' + SG.nameColor(user.name), true).update();
```
Comment (HoeenCoder) [inline spacialgaze-plugins/icons.js]: If were using Monitor.adminlog everywhere else why not here?

Comment (ghost) [reply spacialgaze-plugins/icons.js]: ~~laziness~~ sorry

**Comment by HoeenCoder** [comment]
Looks good, test it before merging it.

**Comment by ghost** [comment]
Ok

**Comment by ghost** [comment]
Will test tonight, cant do it rn cus camp

**Comment by HoeenCoder** [comment]
Any update on tests for this?

**Comment by ghost** [comment]
Monitor.AdminLog is not a function.

As such, the premise of using monitor.AdminLog is a huge lie.

**Comment by DeathlyPlays** [comment]
I thought monitor.AdminLog didn't sound familiar but I didn't want to try correcting anyone with more experience haha

**Comment by HoeenCoder** [comment]
Check your spelling:
(eval)
```
>> Monitor.adminlog
<< function (text) {
this.notice(text);
if (Rooms('upperstaff')) {
Rooms('upperstaff').add('|c|~|' + text).update();
}
}
```

https://github.com/HoeenCoder/SpacialGaze/blob/master/monitor.js#L76

**Comment by ghost** [comment]
fuck...

**Comment by DeathlyPlays** [comment]
Thought I might just do this... #2 this solves the setting issue sorta ._.

**Comment by DeathlyPlays** [comment]
I'd personally prefer to make it actually deny setting it, if one already exists?

**Comment by ghost** [comment]
Fixed the miscapitalization

**Comment by DeathlyPlays** [comment]
Testing today unless @Gligar otherwise does (I'll try anyways)

**Comment by ghost** [comment]
Ok

**Comment by DeathlyPlays** [comment]
Tested and works

**Pull request merged**